### PR TITLE
Enhancements to allow editing a timer's autotimer

### PR DESCRIPTION
**Pull request by SimonCapewell**
1. Provide a method to edit the autotimer given a timer.
1. persisting the autotimer's `id` in autotimer.xml
1. storing `autoTimerId` on each timer that it creates
1. Also provide a method to create an autotimer from a timer

Note that 3 requires changes in enigma2's [RecordTimer.py](https://github.com/SimonCapewell/enigma2/commit/6ef4cb2cc4fc8abeda26f35aa649f6921454f7f5#diff-0f321e0c10bf545be5c64a41d09d6732) to save the new attribute to timer.xml

This is a companion piece to go with the work in https://github.com/SimonCapewell/enigma2/tree/epg-custom-buttons to enable enter editing an autotimer from the EPG screens in the same way you can edit a standard timer.

### CODE REVIEW

File: autotimer/src/AutoTimerEditor.py [hunk 1]
```diff
@@ -1539,6 +1539,42 @@ def addAutotimerFromEvent(session, evt = None, service = None):
 		[]			# Proposed tags
 	)
 
+def addAutotimerFromTimer(session, timer):
+	from AutoTimerComponent import preferredAutoTimerComponent
+	from AutoTimerImporter import AutoTimerImporter
+	from plugin import autotimer
+
+	autotimer.readXml()
```
Comment (jbleyel) [inline autotimer/src/AutoTimerEditor.py:1547]: This is not needed because the list is persistent.

File: autotimer/src/AutoTimerEditor.py [hunk 2]
```diff
@@ -1598,6 +1634,16 @@ def addAutotimerFromService(session, service = None):
 		tags		# Proposed tags
 	)
 
+def editAutotimerFromTimer(session, timer):
+	from plugin import autotimer
+
+	autotimer.readXml()
```
Comment (jbleyel) [inline autotimer/src/AutoTimerEditor.py:1640]: This is not needed because the list is persistent.

File: autotimer/src/AutoTimerEditor.py (hunk 1 above)
Comment (SimonCapewell) [reply autotimer/src/AutoTimerEditor.py:1547]: I'd be wary of removing this as the 4 other access functions do the same thing - I copied their approach. If `autotimer.readXml` really isn't needed, we should consider tidying them up too.

**Comment by jbleyel** [comment]
Do we have a problem if RecordTimer.py has not been changed?

**Comment by SimonCapewell** [comment]
> Do we have a problem if RecordTimer.py has not been changed?
No problems. The new code is backwards compatible: `autoTimerId` is set directly on a new timer, rather than calling the constructor. The changes in RecordTimer are to save and load `autoTimerId` to _timer.xml_

**Pull request merged**